code/game/objects/effects/decals/Cleanable/humans.dm

```diff
@@ -57,6 +90,80 @@ GLOBAL_LIST_EMPTY(splatter_cache)
 	color = adjust_brightness(color, -50)
 	amount = 0
 
+/obj/effect/decal/cleanable/blood/Crossed(atom/movable/O)
+	. = ..()
+	if(ishuman(O) && !gravity_check)
+		bloodyify_mob(O)
+
+/obj/effect/decal/cleanable/blood/proc/splat(atom/AT)
+	if(gravity_check) //only floating blood can splat :C
+		return
+	var/turf/T = get_turf(AT)
+	if(loc != T)
+		forceMove(T) //move to the turf to splatter on
+	animate(src) //stop floating
+	gravity_check = TRUE
+	icon = initial(icon)
+	icon_state = weightless_image.icon_state
+	layer = initial(layer)
+	plane = initial(plane)
+	update_icon()
+
+/obj/effect/decal/cleanable/blood/Process_Spacemove(movement_dir)
+	if(gravity_check)
+		return 1
+
+	if(has_gravity(src))
+		if(!gravity_check)
+			splat(get_step(src, movement_dir))
+		return 1
+
+	if(pulledby && !pulledby.pulling)
+		return 1
+
+	if(throwing)
+		return 1
+
+	return 0
+
+
+/obj/effect/decal/cleanable/blood/Bump(atom/A, yes)
+	if(gravity_check)
+		return ..()
+	if(iswallturf(A) || istype(A, /obj/structure/window))
+		splat(A)
+		return
+	else if(A.density)
+		splat(get_turf(A))
+		return
+
+	..()
+
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		bloodyify_mob(H)
+		return
+
+/obj/effect/decal/cleanable/blood/proc/bloodyify_mob(mob/M)
+	if(!ishuman(M))
+		return
```

Comment:
I would probably just remove this check here and typecast the parameter to be a human. You already ishuman() check mobs before passing them to this proc. Let it runtime otherwise as a reminder for other contributors to properly use this proc